operations/configuration_management/CFS_Key_Management.md

```diff
@@ -0,0 +1,33 @@
+## CFS Key Management and Permission Denied Errors
+
+Configuration Framework Service \(CFS\) manages it's own keys separate from keys for communication between CFS and the components or images that it is configuring.  These are separate from the keys used by users and should not need to be managed.
+
+### Permission Denied Errors
+
+If Ansible is unable to connect with it's target and fails with an `Unreachable - Permission denied` error the first place to check is the `cfs-state-reporter` on the target node.
+
+Every booted node should be running a copy of cfs-state-reporter.  This service is responsible for pulling down the public key.  To check the status of this service, ssh to the node that CFS can't communicate with, and run `systemctl status cfs-state-reporter`. 
+
+The cfs-state-reporter should be complete, but report success.  Any other state
+can be an indication of a problem.
+
+```
+ncn-m001:~ # systemctl status cfs-state-reporter
+● cfs-state-reporter.service - cfs-state-reporter reports configuration level of the system
+   Loaded: loaded (/usr/lib/systemd/system/cfs-state-reporter.service; enabled; vendor preset: disabled)
+   Active: inactive (dead) since Wed 2022-01-19 18:53:45 UTC; 1s ago
+  Process: 678311 ExecStart=/usr/bin/python3 ${MODULEFLAG} ${MODULENAME} (code=exited, status=0/SUCCESS)
+ Main PID: 678311 (code=exited, status=0/SUCCESS)
+ ```
+
+If cfs-state-reporter is complete but has failed, it can safely be restarted with `systemctl restart cfs-state-reporter`, although any log messages in the status should be noted in case they are needed for later debugging.
+
+```
+ncn-m001:~ # systemctl status cfs-state-reporter
+● cfs-state-reporter.service - cfs-state-reporter reports configuration level of the system
+   Loaded: loaded (/usr/lib/systemd/system/cfs-state-reporter.service; enabled; vendor preset: disabled)
+   Active: failed (Result: exit-code) since Thu 2022-01-13 14:41:57 UTC; 6 days ago
+  Process: 14849 ExecStart=/usr/bin/python3 ${MODULEFLAG} ${MODULENAME} (code=exited, status=1/FAILURE)
+ Main PID: 14849 (code=exited, status=1/FAILURE)
+ ```
+cfs-state-reporter may also still be in a running state.  In this case it is likely waiting on another either to authenticate or pull down the ssh key.  The service can safely be restarted as with the failure case, but this is less likely to be successful.  If the log messages indicate problems communicating with Spire, checking the health of the Spire service on the node is the next step.  If there are errors indicating failure to communicate with bss or the metadata service, check the health of both `BSS` with `kubectl -n services logs deployment/cray-bss -c cray-bss` and `cfs-trust` with `kubectl -n services logs deployment/cfs-trust -c cfs-trust`
```

Comment:
We should also add more information on how to check the health of Spire. Not sure if there is another procedure we can point to for that. Maybe /operations/spire/Troubleshoot_Spire_Failing_to_Start_on_NCNs.md?

Reply:
added